content/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-php-project-for-codespaces.md

```diff
@@ -0,0 +1,209 @@
+
+1. Choose the version of PHP you want to use for your project. In this case, select the version marked "(default)."
+
+   <!-- ![Screenshot of the "Add Dev Container Configuration Files" dropdown, showing a variety of PHP versions, including "8.2 (default)."](/assets/images/help/codespaces/add-php-version.png) -->
+
+1. A list of additional features you can install is displayed. <!-- We'll install JSHint, a code quality tool for detecting errors in JavaScript code. To install this tool, type `js`, select `JSHint (via npm)`, then click **OK**. -->
+
+   <!-- ![Screenshot of the "Add Dev Container Configuration Files" dropdown, showing "js" in the text box and "JSHint (via npm)" in the dropdown list.](/assets/images/help/codespaces/add-jshint-config.png) -->
+
+{% data reusables.codespaces.overwrite-devcontainer-config %}
+{% data reusables.codespaces.details-of-devcontainer-config %}
+
+```json
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/php
+{
+  "name": "PHP",
+  // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+  "image": "mcr.microsoft.com/devcontainers/php:0-8.2",
+
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  // "features": {},
+
+  // Configure tool-specific properties.
+  "customizations": {
+    // Configure properties specific to VS Code.
+    "vscode": {
+      "settings": {},
+      "extensions": [
+        "streetsidesoftware.code-spell-checker"
+      ]
+    }
+  },
+
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  // "forwardPorts": [8000],
+
+  // Use 'portsAttributes' to set default properties for specific forwarded ports. More info: https://code.visualstudio.com/docs/remote/devcontainerjson-reference.
+  "portsAttributes": {
+    "8000": {
+      "label": "Hello Remote World",
+      "onAutoForward": "notify"
+    }
+  }
+
+  // Use 'postCreateCommand' to run commands after the container is created.
+  // "postCreateCommand": "sudo chmod a+x \"$(pwd)\" && sudo rm -rf /var/www/html && sudo ln -s \"$(pwd)\" /var/www/html",
+
+  // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
+  // "remoteUser": "root"
+}
+```
+
+{% data reusables.codespaces.devcontainer-properties-1 %}
+{% data reusables.codespaces.devcontainer-properties-2 %}
+{% data reusables.codespaces.additional-container-config %}
+
+## Step 3: Modify your devcontainer.json file
+
+With your dev container configuration added and a basic understanding of what everything does, you can now make changes to customize your environment further. In this example, you'll add properties that will:
+* Run `composer install`, after the dev container is created, to install the dependencies listed in the `composer.json` file.
```

Comment:
I don't follow the reference to a `composer.json` file here. There is no such file in the codespace.